src/software/new_geom/cubic_bezier_spline2d.h

```diff
@@ -0,0 +1,109 @@
+#pragma once
+
+#include "software/new_geom/bezier_curve2d.h"
+#include "software/new_geom/ray.h"
+#include "software/new_geom/spline2d.h"
+
+/**
+ * A Cubic Bezier spline is a C2 continuous spline that is formed by a series of bezier
+ * curves.
+ *
+ * Some useful resources:
+ * https://www.ibiblio.org/e-notes/Splines/Intro.htm
+ * https://www.ibiblio.org/e-notes/Splines/b-int.html
+ */
+class CubicBezierSpline2d : public Spline2d
+{
+   public:
+    CubicBezierSpline2d() = delete;
+
+    /**
+     * Create a CubicBezierSpline2d that:
+     *  - starts from `start` and is tangent to `start_vector` at the start
+     *  - ends at `end` and is tangent to `end_vector` at the end
```

Comment:
nit: can we add a blank line between the description and `@params`?

Reply:
yep, changed